New theme tasks merge request (subtask of item in main repo)

br2490 opened the pull request:
@see: https://github.com/BarnardArchives/digitalcollections.barnard.edu/issues/74

Commit:
Update to pagination for new theme.

Commit:
Update to pagination for new theme.

Commit:
Add pagination javascript
Update lastPage
Update logic (because math. they said study hard. I should've listened.)

tag: @MarthaTenney
Reference: https://github.com/BarnardArchives/digitalcollections.barnard.edu/issues/74

Commit:
Add pagination javascript
Update lastPage
Update logic (because math. they said study hard. I should've listened.)

tag: @MarthaTenney
Reference: https://github.com/BarnardArchives/digitalcollections.barnard.edu/issues/74

Commit:
Update to Search result heading.

Commit:
Remove incorrect data from comment.

Commit:
Update and push for page heading chnges

elanaalt reviewed (commented):
This is so great! I did notice a small issue, though. Page 31 is shown as the last page most of the time. 
<img width="549" alt="Screen Shot 2019-09-09 at 5 48 41 PM" src="https://user-images.githubusercontent.com/43379767/64568972-29824080-d32a-11e9-94af-dfdbc66b8899.png">

However, when you click on page 31, it take you to page 32...
<img width="441" alt="Screen Shot 2019-09-09 at 5 50 29 PM" src="https://user-images.githubusercontent.com/43379767/64569062-5fbfc000-d32a-11e9-9245-2e6af39510f0.png">

Commit:
Edits for https://github.com/BarnardArchives/digitalcollections.barnard.edu/issues/75

br2490 commented:
@elanaalt Thank you for that catch!!! I didn't account for the 0 base of arrays. Updated and ready for review

Commit:
Update pagination ellipsis for where active page is 4.

https://github.com/BarnardArchives/digitalcollections.barnard.edu/issues/74

Commit:
Breakout new code into own files.

Commit:
Solr search heading to read "<result count> Search results"
for: https://github.com/BarnardArchives/digitalcollections.barnard.edu/issues/75

elanaalt commented:
This is great, thanks @br2490!!
